### PR TITLE
DROOLS-4195: [DMN Designer] PMML: Update Document value when Import alias changes

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/DROOLS-4195

### CODE REVIEW

**Comment by manstis** [comment]
@mbiarnes Anything I should be doing to get PRs to be built by Jenkins!?

Jenkins please retest this.

ok to test.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/included/imports/persistence/DMNIncludedModelHandler.java [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    private void updateDRGElementsVariableAndId(final String oldModelName,
+                                                final String newModelName) {
+        dmnGraphUtils
+                .getDRGElements()
+                .forEach(drgElement -> {
+
+                    final Id oldId = drgElement.getId();
+
+                    if (oldId.getValue().startsWith(oldModelName)) {
+                        final String newId = oldId.getValue().replaceAll("^" + oldModelName, newModelName);
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/included/imports/persistence/DMNIncludedModelHandler.java]: Do you think it is worth to cover also this case?
Screenshot from 2019-06-26 10-59-40

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/included/imports/persistence/PMMLIncludedModelHandler.java [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+            final Context context = entry.getValue();
+            for (final ContextEntry contextEntry : context.getContextEntry()) {
+                if (Objects.equals(LiteralExpressionPMMLDocument.VARIABLE_DOCUMENT, contextEntry.getVariable().getName().getValue())) {
+                    final Expression expression = contextEntry.getExpression();
+                    if (expression instanceof IsLiteralExpression) {
+                        final IsLiteralExpression ile = (IsLiteralExpression) expression;
+                        if (Objects.nonNull(ile.getText())) {
+                            final Text text = ile.getText();
+                            if (Objects.equals(quotedOldModelName, text.getValue())) {
+                                clearContextValues(context);
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/included/imports/persistence/PMMLIncludedModelHandler.java]: From my point of view `clearContextValues` do basically same checks as are done in `destroy`. I think we should remove `clearContextValues` and use just `text.setValue("")` at this place. Am I missing something?

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/included/imports/persistence/PMMLIncludedModelHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,493 @@
...
+        final List<Decision> decisions = setupDecisionWithTopLevelFunction();
+
+        handler.update(DOCUMENT_NAME_ORIGINAL, DOCUMENT_NAME_UPDATED);
+
+        assertTopLevelLiteralExpression(decisions.get(0));
+        assertTopLevelFunctionDefinition(decisions.get(1),
+                                         DOCUMENT_NAME_UPDATED,
+                                         DOCUMENT_NAME_ORIGINAL);
+        assertTopLevelFunctionDefinition(decisions.get(2),
+                                         "do-not-change",
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/included/imports/persistence/PMMLIncludedModelHandlerTest.java]: Could we use something like "document-not-affected" ? Same for other tests.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/included/imports/persistence/PMMLIncludedModelHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,493 @@
...
+                                         "do-not-change");
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/included/imports/persistence/PMMLIncludedModelHandlerTest.java]: Could we use something like "model-not-affected" ? Same for other tests.

**Comment by jomarko** [comment]
### Manual Review
Passed :heavy_check_mark:

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/included/imports/persistence/PMMLIncludedModelHandlerTest.java (hunk 3 above)
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/included/imports/persistence/PMMLIncludedModelHandlerTest.java]: Done.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/included/imports/persistence/PMMLIncludedModelHandlerTest.java (hunk 4 above)
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/included/imports/persistence/PMMLIncludedModelHandlerTest.java]: Done.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/included/imports/persistence/PMMLIncludedModelHandler.java (hunk 2 above)
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/included/imports/persistence/PMMLIncludedModelHandler.java]: @jomarko Ah well.. they're not doing quite the same thing.. here I'm finding a `Context` that contains a variable for PMMLs "document" property; then when found `clearContextValues(..)` clears the value of _ALL_ values for the identified `Context`'s `ContextEntry`'s. i.e. clearing the value for both "document" and "model".

Comment (jomarko) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/included/imports/persistence/PMMLIncludedModelHandler.java]: Michael 1
Jozef 0

Thank you for answer.

**Comment by jomarko** [comment]
@manstis please push, will approve then.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/included/imports/persistence/DMNIncludedModelHandler.java [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+
+            if (nameId != null) {
+                commandBuilder.addCommand(canvasCommandFactory.updatePropertyValue(node, nameId, newName));
+            }
+        }
+
+        return commandBuilder.build();
+    }
+
+    Node getNode(final Object definition) {
```
[the remark this answers is not in the document]
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/included/imports/persistence/DMNIncludedModelHandler.java]: @jomarko Similar but not identical. This one finds a `Node` where the `Definition` matches, whereas `BaseGrid.findDomainObjectNodeByDomainObject(..)` finds a `Node` where the `DomainObject` matches. All that could be moved to a common utility class would be:-
```
StreamSupport
    .stream(graph.nodes().spliterator(), false)
    .filter((Node node) -> node.getContent() instanceof Definition)
```
Which seems pretty pointless. Would you like me to still do that?

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/included/imports/persistence/DMNIncludedModelHandler.java (hunk 1 above)
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/included/imports/persistence/DMNIncludedModelHandler.java]: I only renamed this class; but have extended an existing test to include that code path.

**Comment by manstis** [comment]
@jomarko Updates made in new commit. Thank-you.

**Comment by manstis** [comment]
@karreiro Could you please take a look when you get a moment free. Thanks.

**Pull request merged**